### PR TITLE
GatewayController move endpoints to gateway

### PR DESCRIPTION
**Pull request by RadoslavMonov**

### CODE REVIEW

File: .idea/compiler.xml [hunk 1]
```diff
@@ -2,6 +2,7 @@
 <project version="4">
   <component name="CompilerConfiguration">
     <annotationProcessing>
+      <profile default="true" name="Default" enabled="true" />
```
Comment (SavageStan) [inline .idea/compiler.xml]: Why is this necessary?

File: src/main/java/com/monov/cloud/gateway/CloudGatewayApplication.java [hunk 2]
```diff
@@ -2,7 +2,10 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.cloud.client.loadbalancer.LoadBalanced;
 import org.springframework.cloud.netflix.eureka.EnableEurekaClient;
+import org.springframework.context.annotation.Bean;
+import org.springframework.web.client.RestTemplate;
 
 @EnableEurekaClient
```
Comment (SavageStan) [inline src/main/java/com/monov/cloud/gateway/CloudGatewayApplication.java]: You probably no longer need this @EnableEurekaClient annotation since we decided to stop using the gateway as a router only. Also optimize your imports via Ctrl + Alt + O

File: .idea/compiler.xml (hunk 1 above)
Comment (SavageStan) [reply .idea/compiler.xml]: We should not be committing changes to files in .idea to our version control.

File: src/main/java/com/monov/cloud/gateway/data/StudentOrCourseIds.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.monov.cloud.gateway.data;
+
+import lombok.Data;
+
+import java.util.List;
+
+
+@Data
+public class StudentOrCourseIds {
```
Comment (SavageStan) [inline src/main/java/com/monov/cloud/gateway/data/StudentOrCourseIds.java:9]: I would name this ItemIds. In the future you might find a use for this class for another type of entity.

File: .idea/compiler.xml (hunk 1 above)
Comment (RadoslavMonov) [reply .idea/compiler.xml]: done

File: src/main/java/com/monov/cloud/gateway/CloudGatewayApplication.java (hunk 2 above)
Comment (RadoslavMonov) [reply src/main/java/com/monov/cloud/gateway/CloudGatewayApplication.java]: done

File: src/main/java/com/monov/cloud/gateway/data/StudentOrCourseIds.java (hunk 3 above)
Comment (RadoslavMonov) [reply src/main/java/com/monov/cloud/gateway/data/StudentOrCourseIds.java:9]: done

File: src/main/java/com/monov/cloud/gateway/dto/Course.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import lombok.NoArgsConstructor;
+
+import java.util.List;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class Course {
+
+    private Long courseId;
```
Comment (SavageStan) [inline src/main/java/com/monov/cloud/gateway/dto/Course.java:14]: Whenever you have a model called for example Course, the id field does not need to be called courseId since it is implied that a field with name Id would be the id of the Course model. Look at your other models as well, they might have the same thing. 

Also I see that you are placing these models in a DTO package. If you intend them to be DTOs then name them CourseDTO etc. Since these DTO classes are going to be used in more than 1 microservice (their own microservice and the gateway) it might be good to move them outside into a common library/project to avoid duplicate classes.

File: src/main/java/com/monov/cloud/gateway/dto/StudentOrCourseIds.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.List;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class StudentOrCourseIds {
```
Comment (SavageStan) [inline src/main/java/com/monov/cloud/gateway/dto/StudentOrCourseIds.java]: Seems like you already have this model inside a different package?

File: src/main/java/com/monov/cloud/gateway/service/CourseGatewayService.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+@Service
+public class CourseGatewayService {
+    public static final String COURSE_SERVICE_URL = "http://localhost:9002/courses";
```
Comment (SavageStan) [inline src/main/java/com/monov/cloud/gateway/service/CourseGatewayService.java:15]: Extract this into a config file (.properties or .yaml) and use the @Value annotation to inject it here.

File: src/main/java/com/monov/cloud/gateway/service/StudentGatewayService.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+@Service
+public class StudentGatewayService {
+
+    public static final String STUDENT_SERVICE_URL = "http://localhost:9001/students";
```
Comment (SavageStan) [inline src/main/java/com/monov/cloud/gateway/service/StudentGatewayService.java:16]: Move this to .properties / .yaml

File: src/main/java/com/monov/cloud/gateway/service/StudentGatewayService.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        return restTemplate.postForObject(String.format("%s/%d/%d",STUDENT_SERVICE_URL,studentId,courseId),null,
+                Student.class);
+    }
+
+    public List<Student> getStudentsByCourseId(Long courseId) {
+        return Arrays.asList(Objects.requireNonNull(restTemplate.getForObject(String.format("%s/course/%d", STUDENT_SERVICE_URL, courseId),
+                Student[].class)));
+    }
+
+//    public List<Course> getCoursesForStudentById(Long studentId) {
```
Comment (SavageStan) [inline src/main/java/com/monov/cloud/gateway/service/StudentGatewayService.java:45]: Commented out code should not be committed (except in very specific cases)

File: src/main/resources/application.yaml [hunk 9]
```diff
@@ -4,19 +4,7 @@ server:
...
-          uri: lb://STUDENT-SERVICE
-          predicates:
-            - Path=/students/**
-        - id: COURSE-SERVICE
-          uri: lb://COURSE-SERVICE
-          predicates:
-            - Path=/courses/**
 
-#lb://STUDENT-SERVICE = load balanced://application.name
 eureka:
```
Comment (SavageStan) [inline src/main/resources/application.yaml:8]: You probably no longer need these eureka properties.

File: target/classes/application.yaml [hunk 10]
```diff
@@ -4,19 +4,7 @@ server:
...
-          uri: lb://STUDENT-SERVICE
-          predicates:
-            - Path=/students/**
-        - id: COURSE-SERVICE
-          uri: lb://COURSE-SERVICE
-          predicates:
-            - Path=/courses/**
 
-#lb://STUDENT-SERVICE = load balanced://application.name
 eureka:
```
Comment (SavageStan) [inline target/classes/application.yaml:8]: You probably no longer need these eureka properties.

**Review by SavageStan — COMMENTED** [review]
Added some comments.

File: src/main/java/com/monov/cloud/gateway/dto/Course.java (hunk 4 above)
Comment (RadoslavMonov) [reply src/main/java/com/monov/cloud/gateway/dto/Course.java:14]: Fixes will be applied in DTOs branch's PR

File: src/main/java/com/monov/cloud/gateway/dto/StudentOrCourseIds.java (hunk 5 above)
Comment (RadoslavMonov) [reply src/main/java/com/monov/cloud/gateway/dto/StudentOrCourseIds.java]: Fixes will be applied in DTOs branch's PR

File: src/main/java/com/monov/cloud/gateway/service/CourseGatewayService.java (hunk 6 above)
Comment (RadoslavMonov) [reply src/main/java/com/monov/cloud/gateway/service/CourseGatewayService.java:15]: Fixes will be applied in DTOs branch's PR

File: src/main/java/com/monov/cloud/gateway/service/StudentGatewayService.java (hunk 8 above)
Comment (RadoslavMonov) [reply src/main/java/com/monov/cloud/gateway/service/StudentGatewayService.java:45]: Fixes will be applied in DTOs branch's PR

File: src/main/java/com/monov/cloud/gateway/service/StudentGatewayService.java (hunk 7 above)
Comment (RadoslavMonov) [reply src/main/java/com/monov/cloud/gateway/service/StudentGatewayService.java:16]: Fixes will be applied in DTOs branch's PR

File: src/main/resources/application.yaml (hunk 9 above)
Comment (RadoslavMonov) [reply src/main/resources/application.yaml:8]: Fixes will be applied in DTOs branch's PR

File: target/classes/application.yaml (hunk 10 above)
Comment (RadoslavMonov) [reply target/classes/application.yaml:8]: Fixes will be applied in DTOs branch's PR

**Review by SavageStan — APPROVED** [review]
Approving in order to avoid major merge conflicts despite it being a bad practice since there are unaddressed comments. Will follow up with these requested changes in next PR. I take responsibility to look for the changes.
